src/GraphQL/GraphQLExtensions.cs

```diff
@@ -502,15 +509,59 @@ public static bool DoTypesOverlap(IGraphType typeA, IGraphType typeB)
 
         private static readonly NullValue _null = new NullValue();
 
+        /// <summary>
+        /// Returns a value indicating whether the provided value is a valid default value
+        /// for the specified input graph type.
+        /// </summary>
+        public static bool IsValidDefault(this IGraphType type, object value)
+        {
+            if (type == null)
+                throw new ArgumentNullException(nameof(type));
+
+            if (type is NonNullGraphType nonNullGraphType)
+            {
+                return value == null ? false : nonNullGraphType.ResolvedType.IsValidDefault(value);
+            }
+
+            if (value == null)
+            {
+                return true;
+            }
+
+            if (type is ListGraphType listGraphType)
+            {
+                if (!(value is IEnumerable list))
```

Comment:
Wait. What about coersion from a value to single-item list ?
```graphql
type Query {
  field1(arg: [String] = "aaa")
  field2(arg: [String] = ["aaa"])
}
```

field1 and field2 are effectively the same as I think.

Reply:
If so please add one test to verify.

Reply:
Also see this comment about support value as a single-item list:
```
 // Convert IEnumerable to GraphQL list. If the GraphQLType is a list, but
            // the value is not an IEnumerable, convert the value using the list's item type.
            if (type is ListGraphType listType)
```

Reply:
#2334

Reply:
ok